File: Semestrul 1/Fundamentele Programarii/extra/Magazin de biciclete/repository/repository.py
from domain.produs import Produs

class Repository:

    def __init__(self,  filename):
        self.__filename = filename

    def load_from_file(self):
        produse = []
        with open(self.__filename, 'r') as f:
            lines = f.readlines()
            for line in lines:
                line = line.strip()
                id_bicicleta, tip, pret = line.split(", ")
                produs = Produs(int(id_bicicleta), tip, float(pret))
                produse.append(produs)
        f.close()
        return produse

    def save_to_file(self, produse):
        with open(self.__filename, 'w') as f:
            for el in produse:
                string = str(el.get_id_bicicleta()) + ', ' + str(el.get_tip()) + ', ' + str(el.get_pret()) + '\n'
                f.write(string)
        f.close()

    def get_all(self):
        return self.load_from_file()

    def delete(self, id_bicicleta):
        produse = []
        for el in self.load_from_file():
            if el.get_id_bicicleta() != id_bicicleta:
                produse.append(el)
        self.save_to_file(produse)

File: Semestrul 2/Programare orientata obiect/Pregatire_Sesiune/App/repository.cpp
#pragma once
#include <sstream>
#include "repository.h"

void Repository::loadFromFile() {
    all.clear();
    ifstream fin(fileName);
    string line;
    while(getline(fin, line)) {
        int id;
        string titlu, tip;
        double pret;
        string curent;
        stringstream linestream(line);
        int nr = 0;
        while(getline(linestream, curent, ',')) {
            if(nr == 0) {
                id = stoi(curent);
            }
            if(nr == 1) {
                titlu = curent;
            }
            if(nr == 2) {
                tip = curent;
            }
            if(nr == 3) {
                pret = stod(curent);
            }
            nr++;
        }
        all.push_back(Produs(id, titlu, tip, pret));
    }
    fin.close();
}

void Repository::saveToFile(vector<Produs> produse) {
    ofstream fout(fileName);
    for(auto &p : produse) {
        fout<<p.get_id()<<","<<p.get_titlu()<<","<<p.get_tip()<<","<<p.get_pret()<<endl;
    }
    fout.close();
}

void Repository::adaugare(Produs p) {
    all.push_back(p);
    saveToFile(all);
}
